components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/listener/SCIMUserOperationListener.java

```diff
@@ -178,48 +182,75 @@ public boolean doPreSetUserClaimValueWithID(String userID, String claimURI, Stri
             // Validate whether claim update request is for a provisioned user.
             validateClaimUpdate(getUsernameFromUserID(userID, userStoreManager));
         }
-        // Validate dob value against the regex.
-        validateDateOfBirthClaimValue(claimURI, claimValue, userStoreManager);
+        // Validate dob and mobile number value against the regex.
+        validateClaimValue(claimURI, claimValue, userStoreManager);
         // Validate if the groups are updated.
         validateUserGroupClaim(userID, claimURI, claimValue, userStoreManager);
         return true;
     }
 
-    private void validateDateOfBirthClaimValue(String claimURI, String claimValue, UserStoreManager userStoreManager)
+    private void validateClaimValue(String claimURI, String claimValue, UserStoreManager userStoreManager)
```

Comment:
Better to add a comment mentioning which claims are validated through this method. Method name is generic, but we're only validating mobile and dob claims. WDYT?

Reply:
fixed in e09571c